File: 프로그래머스/2/87946. 피로도/피로도.py
def solution(k, dungeons):
    from itertools import permutations as p
    r,l=[],len(dungeons)
    for e in p(range(l)):
        t,s=k,0
        for i in list(e):
            if t>=dungeons[i][0]:
                t-=dungeons[i][1]
                s+=1
        r+=[s]
        if s==l:
            break  
    return max(r)

File: 프로그래머스/2/87946. 피로도/피로도.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int k, vector<vector<int>> dungeons) {
    int r=0,tmp,cnt;
    vector<int> idx;
    for(int i=0;i<dungeons.size();i++) {
        idx.push_back(i);
    }
    do{
        tmp=k,cnt=0;
        for(int i=0;i<idx.size();i++) {
            if(tmp>=dungeons[idx[i]][0]) {
                cnt++;
                tmp-=dungeons[idx[i]][1];
            }
        }
        r=max(r,cnt);
    } while(next_permutation(idx.begin(),idx.end()));
    return r;
}